First build on Windows causes file not found error for core.meta.slang.h

Hi,

This is my first time with slang, and I have set up a MSVC2022 project that uses Slang to convert Slang shaders to platform-specific shaders (HLSL, SPIRV-V and Metal namely).

My CMake setup is as follows:

```cmake
option(SLANG_ENABLE_SLANG_RHI "" OFF)
option(SLANG_EMBED_CORE_MODULE_SOURCE "" ON)
option(SLANG_EMBED_CORE_MODULE "" ON)
option(SLANG_ENABLE_PREBUILT_BINARIES "" OFF)
option(SLANG_ENABLE_REPLAYER "" OFF)
option(SLANG_ENABLE_GFX "" OFF)
option(SLANG_ENABLE_SLANGD "" OFF)
option(SLANG_ENABLE_SLANGC "" OFF)
option(SLANG_ENABLE_SLANGI "" OFF)
option(SLANG_ENABLE_SLANGRT "" OFF)
option(SLANG_ENABLE_SLANG_GLSLANG "" OFF)
option(SLANG_ENABLE_DXIL "" OFF)
option(SLANG_ENABLE_OPTIX "" OFF)
option(SLANG_ENABLE_TESTS "" OFF)
option(SLANG_ENABLE_EXAMPLES "" OFF)
option(SLANG_ENABLE_SPLIT_DEBUG_INFO "" OFF)
option(SLANG_ENABLE_RELEASE_DEBUG_INFO "" OFF)
set(SLANG_LIB_TYPE STATIC CACHE STRING "")
add_subdirectory(3rdparty/slang)
```

Then I simply add slang as a dependency to one of my targets:

```cmake
target_link_libraries(${MODULE}
    PUBLIC
    slang
    core
    ...
)
```

The first build in MSVC2022 via `F7` or `CTRL+B` causes an include file not found error for `core.meta.slang.h` which is then resolved by a subsequent build attempt:

````
43>------ Build started: Project: slang-no-embedded-core-module-source, Configuration: Debug x64 ------
44>------ Build started: Project: slang-embedded-core-module-source, Configuration: Debug x64 ------
44>Generating core-module-meta/core.meta.slang.h, core-module-meta/diff.meta.slang.h, core-module-meta/glsl.meta.slang.h, core-module-meta/hlsl.meta.slang.h
43>Building Custom Rule C:/Users/<user>/Desktop/Project/3rdparty/slang/source/slang-core-module/CMakeLists.txt
43>slang-embedded-core-module-source.cpp
44>Building Custom Rule C:/Users/<user>/Desktop/Project/3rdparty/slang/source/slang-core-module/CMakeLists.txt
44>slang-embedded-core-module-source.cpp
43>C:\Users\<user>\Desktop\Project\ProjectCore\3rdparty\slang\source\slang-core-module\slang-embedded-core-module-source.cpp(360,10): error C1083: Cannot open include file: 'core.meta.slang.h': No such file or directory
43>Done building project "slang-no-embedded-core-module-source.vcxproj" -- FAILED.
````

Did I do anything wrong?

Thanks

--- a/source/slang-core-module/CMakeLists.txt
+++ b/source/slang-core-module/CMakeLists.txt
@@ -69,8 +69,6 @@ set(core_module_source_common_args
     generated
     EXPLICIT_SOURCE
     ./slang-embedded-core-module-source.cpp
-    EXTRA_COMPILE_DEFINITIONS_PRIVATE
-    SLANG_EMBED_CORE_MODULE_SOURCE
     EXPORT_MACRO_PREFIX
     SLANG
     EXPORT_TYPE_AS

--- a/source/slang-core-module/slang-embedded-core-module-source.cpp
+++ b/source/slang-core-module/slang-embedded-core-module-source.cpp
@@ -394,6 +394,7 @@ ComPtr<ISlangBlob> Session::getAutodiffLibraryCode()
 
 ComPtr<ISlangBlob> Session::getGLSLLibraryCode()
 {
+#if SLANG_EMBED_CORE_MODULE_SOURCE
     if (!glslLibraryCode)
     {
         const String path = getCoreModulePath();
@@ -401,6 +402,7 @@ ComPtr<ISlangBlob> Session::getGLSLLibraryCode()
 #include "glsl.meta.slang.h"
         glslLibraryCode = StringBlob::moveCreate(sb);
     }
+#endif
     return glslLibraryCode;
 }
 } // namespace Slang

